lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -121,18 +121,27 @@ tokenize(String, Line, Column, Opts) ->
 
   Existing = case lists:keyfind(existing_atoms_only, 1, Opts) of
     {existing_atoms_only, true} -> true;
+    {existing_atoms_only, false} -> false;
     false -> false
   end,
 
   Check = case lists:keyfind(check_terminators, 1, Opts) of
     {check_terminators, false} -> false;
+    {check_terminators, true} -> true;
     false -> true
   end,
 
+  Preserve = case lists:keyfind(preserve_comments, 1, Opts) of
```

Comment:
The names of the variables above are not the best inspiration. Let's call this `PreserveComments` :)